src/ui/Facet/FacetSettings.ts

```diff
@@ -241,6 +251,20 @@ export class FacetSettings extends FacetSort {
     return { element: sortSection, sortItems: sortItems };
   }
 
+  private closePopupOnLoseFocus(el: HTMLElement) {
+    $$(el).on('blur', async () => {
+      const promise = (this.loseFocusPromise = Utils.resolveAfter(15));
```

Comment:
🤔 If I'm reading correctly the popup will try to close itself everytime you blur out of "one section" in the popup (which can contain multiple section).

Why not simply a single "blur" event on the popup container itself (which contains all child section) ? 

Wouldn't this remove pretty much all the logic we have here, and the need to keep a reference to `this. loseFocusPromise` ?

Reply:
I added this because I thought we couldn't detect `blur` on a parent element, but now I found the `focusout` event which seems to accomplish this purpose much better. I'll replace all of this ugly logic with a simple `focusout`.